### PR TITLE
Makes Hulk Attacking Less Snowflakey

### PR DESCRIPTION
**Pull request by Fox-McCloud**
From TG.

Refactors hulk mutation when attacking things. 

Instead of just checking for `if(HULK in mutations)` everywhere in `attack_hand`, we can just call `attack_hulk` on things if you have the hulk mutation.

This carries the balance flag because this dramatically expands just what a hulk can attack, from doors, to walls, to various other structures.

For the vast majority of hulk interactions now, being on harm intent is required; no more accidentally punching walls because you misclicked when hugging someone.

This also fixes the bug with hulk working with TK (allowing you to destroy windows and walls at range).

:cl: Fox McCloud
fix: Hulk mutation no longer works at range (ie: punching/breaking windows/walls at range)
tweak: You must be on harm intent to damage things if you have hulk
add: Having hulk allows you to punch and damage just about anything
/:cl:

### CODE REVIEW

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 1]
```diff
@@ -205,11 +205,12 @@
 /obj/structure/foamedmetal/attack_hand(mob/user)
```
Comment (Fox-McCloud) [inline code/game/objects/effects/effect_system/effects_foam.dm]: This needs to be converted to the new object damage system.

File: code/game/objects/structures/aliens.dm [hunk 2]
```diff
@@ -112,14 +112,6 @@
 	health -= tforce
 	healthcheck()
 
-/obj/structure/alien/resin/attack_hand(mob/living/user)
```
Comment (Fox-McCloud) [inline code/game/objects/structures/aliens.dm]: This needs to be converted to the new object damage system.

File: code/game/objects/structures/window.dm [hunk 3]
```diff
@@ -127,13 +127,8 @@ var/global/wcCommon = pick(list("#379963", "#0d8395", "#58b5c3", "#49e46e", "#8f
...
 		destroy()
 
-
-/obj/structure/window/attack_hand(mob/user as mob)
-	if(HULK in user.mutations)
-		user.say(pick(";RAAAAAAAARGH!", ";HNNNNNNNNNGGGGGGH!", ";GWAAAAAAAARRRHHH!", "NNNNNNNNGGGGGGGGHH!", ";AAAAAAARRRGH!"))
-		user.visible_message("<span class='danger'>[user] smashes through [src]!</span>")
-		destroy()
-	else if(user.a_intent == INTENT_HARM)
+/obj/structure/window/attack_hand(mob/user)
```
Comment (Fox-McCloud) [inline code/game/objects/structures/window.dm]: This needs to be converted to the new object damage system.

File: code/modules/fish/fishtank.dm [hunk 4]
```diff
@@ -313,7 +313,7 @@
 	egg_list.Cut()									//Destroy any excess eggs, clearing the egg_list
 
 /obj/machinery/fishtank/proc/harvest_fish(var/mob/user)
```
Comment (Fox-McCloud) [inline code/modules/fish/fishtank.dm]: This needs to be converted to the new object damage system.

File: code/game/machinery/OpTable.dm [hunk 5]
```diff
@@ -54,13 +54,12 @@
 	if(prob(75))
 		qdel(src)
 
-/obj/machinery/optable/attack_hand(mob/user as mob)
-	if(HULK in usr.mutations)
-		to_chat(usr, text("<span class='notice'>You destroy the table.</span>"))
-		visible_message("<span class='warning'>[usr] destroys the operating table!</span>")
-		src.density = 0
+/obj/machinery/optable/attack_hulk(mob/living/carbon/human/user, does_attack_animation = FALSE)
```
Comment (Fox-McCloud) [inline code/game/machinery/OpTable.dm]: this should *probably* be converted over to the new object damage system; this thing is super snowflakey.

**Comment by Fox-McCloud** [comment]
This is no longer WIP. Only thing this is waiting on is @uraniummeltdown's window refactor PR. Once that's merged in, I'll solve conflicts and this should be ok.

File: code/game/atoms.dm [hunk 6]
```diff
@@ -143,6 +143,12 @@
 /atom/proc/setDir(newdir)
 	dir = newdir
 
+/atom/proc/attack_hulk(mob/living/carbon/human/user, does_attack_animation = FALSE)
+	if(does_attack_animation)
+		user.changeNext_move(CLICK_CD_MELEE)
+		add_attack_logs(user, src, "punched with hulk powers")
```
Comment (tigercat2000) [inline code/game/atoms.dm]: Capitalize `punched` please (consistency)

File: code/game/mecha/mecha.dm [hunk 7]
```diff
@@ -470,19 +470,10 @@
...
-		"<span class='danger'>You hit [name] with all your might. The metal creaks and bends.</span>")
-	else
-		user.do_attack_animation(src, ATTACK_EFFECT_PUNCH)
-		playsound(loc, 'sound/weapons/tap.ogg', 40, 1, -1)
-		user.visible_message("<span class='danger'>[user] hits [name]. Nothing happens</span>","<span class='danger'>You hit [name] with no visible effect.</span>")
-		log_append_to_last("Armor saved.")
-	return
+	user.do_attack_animation(src, ATTACK_EFFECT_PUNCH)
+	playsound(loc, 'sound/weapons/tap.ogg', 40, 1, -1)
+	user.visible_message("<span class='danger'>[user] hits [name]. Nothing happens</span>","<span class='danger'>You hit [name] with no visible effect.</span>")
```
Comment (tigercat2000) [inline code/game/mecha/mecha.dm]: Spacing

File: code/game/mecha/mecha.dm [hunk 8]
```diff
@@ -526,6 +517,16 @@
...
 
+/obj/mecha/hulk_damage()
+	return 15
+
+/obj/mecha/attack_hulk(mob/living/carbon/human/user)
+	. = ..()
+	if(.)
+		check_for_internal_damage(list(MECHA_INT_TEMP_CONTROL, MECHA_INT_TANK_BREACH, MECHA_INT_CONTROL_LOST))
+		log_message("Attack by hulk. Attacker - [user].", 1)
+		add_attack_logs(user, src, "punched with hulk powers")
```
Comment (tigercat2000) [inline code/game/mecha/mecha.dm]: Capitalize here too please.

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 9]
```diff
@@ -186,64 +186,31 @@
 	..()
 	move_update_air(T)
 
+/obj/structure/foamedmetal/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
+	playsound(src.loc, 'sound/weapons/tap.ogg', 100, 1)
```
Comment (tigercat2000) [inline code/game/objects/effects/effect_system/effects_foam.dm]: src.

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 10]
```diff
@@ -186,64 +186,31 @@
...
+		max_integrity = 50
+		obj_integrity = max_integrity
 
 /obj/structure/foamedmetal/attack_hand(mob/user)
 	user.changeNext_move(CLICK_CD_MELEE)
 	user.do_attack_animation(src, ATTACK_EFFECT_PUNCH)
-	if((HULK in user.mutations) || (prob(75 - metal*25)))
-		user.visible_message("<span class='warning'>[user] smashes through \the [src].</span>", "<span class='notice'>You smash through \the [src].</span>")
+	if(prob(75 - metal * 25))
+		user.visible_message("<span class='warning'>[user] smashes through  [src].</span>", "<span class='notice'>You smash through [src].</span>")
```
Comment (tigercat2000) [inline code/game/objects/effects/effect_system/effects_foam.dm]: There's a double space between through and [src]

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 11]
```diff
@@ -186,64 +186,31 @@
...
 		qdel(src)
 	else
 		to_chat(user, "<span class='notice'>You hit the metal foam but bounce off it.</span>")
+		playsound(src.loc, 'sound/weapons/tap.ogg', 100, 1)
```
Comment (tigercat2000) [inline code/game/objects/effects/effect_system/effects_foam.dm]: src.

File: code/game/objects/items/weapons/twohanded.dm [hunk 12]
```diff
@@ -247,6 +247,11 @@
 		return ..()
 	return 0
 
+/obj/item/twohanded/dualsaber/attack_hulk(mob/living/carbon/human/user, does_attack_animation = FALSE)  //In case thats just so happens that it is still activated on the groud, prevents hulk from picking it up
+	if(wielded)
+		to_chat(user, "<span class='warning'>You can't pick up such dangerous item with your meaty hands without losing fingers, better not to!</span>")
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/twohanded.dm]: such a

File: code/modules/mob/living/carbon/alien/humanoid/humanoid_defense.dm [hunk 13]
```diff
@@ -1,16 +1,25 @@
+/mob/living/carbon/alien/humanoid/attack_hulk(mob/living/carbon/human/user, does_attack_animation = FALSE)
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid_defense.dm]: Does this need an attack log?

Comment (Fox-McCloud) [reply code/modules/mob/living/carbon/alien/humanoid/humanoid_defense.dm]: Discussed this; `atom/proc/attack_hulk` should handle this properly.

**Pull request merged**